modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/util.rb

```diff
@@ -80,7 +80,46 @@ def token(user:, configuration:, &)
     end
 
     def error_text_from_response(response)
-      Nokogiri::XML(response.body).xpath("//s:message").text
+      response.xml.xpath("//s:message").text
+    end
+
+    def origin_user_id(caller:, storage:, auth_strategy:)
+      case auth_strategy.key
+      when :basic_auth
+        ServiceResult.success(result: storage.username)
+      when :oauth_user_token
+        origin_user_id = OAuthClientToken.find_by(user_id: auth_strategy.user, oauth_client: storage.oauth_client)&.origin_user_id
```

Comment:
🟡 Since we expect just one record to return you could use [`pick`](https://api.rubyonrails.org/classes/ActiveRecord/Calculations.html#method-i-pick)

```suggestion
        origin_user_id = OAuthClientToken.where(user_id: auth_strategy.user, oauth_client: storage.oauth_client).pick(:origin_user_id)
```